Unicode/ustring.c

```diff
@@ -345,72 +358,79 @@ return( u_copy( s1 ));
 return( pt );
 }
 
-unichar_t *uc_copyn(const char *pt,int len) {
-    unichar_t *res, *rpt;
-
-    if(!pt)
-return((unichar_t *)0);
+unichar_t *uc_copyn(const char *pt, int n) {
+/* Copy n unsigned chars{0..255} to a unichar_t type */
+/* string. If not printable, return an empty string. */
+    unichar_t *res, *rpt, t;
 
+    if ( pt && ++n>0 ) {
 #ifdef MEMORY_MASK
-    if ( (len+1)*sizeof(unichar_t)>=MEMORY_MASK )
-	len = MEMORY_MASK/sizeof(unichar_t)-1;
+	if ( n>MEMORY_MASK/sizeof(unichar_t) )
+	    n = MEMORY_MASK/sizeof(unichar_t);
 #endif
-    res = (unichar_t *) malloc((len+1)*sizeof(unichar_t));
-    for ( rpt=res; --len>=0 ; *rpt++ = *(unsigned char *) pt++ );
-    *rpt = '\0';
-return(res);
+	if ( (res=rpt=(unichar_t *)(malloc(n*sizeof(unichar_t)))) ) {
+	    if ( (--n) ) {
+		while ( --n>=0 ) {
+		    t = (unichar_t)(*(unsigned char *) pt++);
+		    *rpt++ = t & 0xff;
+		}
+	    }
+	    *rpt = 0;
+	    return( res );
+	}
+    }
+    return( calloc(1,sizeof(unichar_t)) );
 }
 
 unichar_t *uc_copy(const char *pt) {
-    unichar_t *res, *rpt;
+/* Copy unsigned char{0..255} string to unichar type */
+/* string. If not printable, return an empty string. */
     int n;
 
-    if(!pt)
-return((unichar_t *)0);
+    if ( pt && (n=strlen(pt))>0 )
+	return( uc_copyn(pt,n) );
 
-    n = strlen(pt);
-#ifdef MEMORY_MASK
-    if ( (n+1)*sizeof(unichar_t)>=MEMORY_MASK )
-	n = MEMORY_MASK/sizeof(unichar_t)-1;
-#endif
-    res = (unichar_t *) malloc((n+1)*sizeof(unichar_t));
-    for ( rpt=res; --n>=0 ; *rpt++ = *(unsigned char *) pt++ );
-    *rpt = '\0';
-return(res);
+    return( calloc(1,sizeof(unichar_t)) );
 }
 
-char *cu_copyn(const unichar_t *pt,int len) {
+char *cu_copyn(const unichar_t *pt, int n) {
```

Comment:
Renaming the parameter from `len` to a letter makes the calling convention less clear.

Reply:
function call uc_... uses n, and both functions are cu_copyn() and uc_copyn()
It seemed to be more consistent to go with n, instead of changing the other function to use len.